### PR TITLE
feat: redux authentication logic

### PR DESCRIPTION
**Pull request by BotsiunDenys**

### CODE REVIEW

File: frontend/src/http/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      originalRequest._isRetry = true;
+      try {
+        const response = await axios.post<AccessToken>(
+          `${API_URL}/auth/refresh/`,
+          { refresh: localStorage.getItem("refresh") }
+        );
+        localStorage.setItem("access", response.data.access);
+        return api.request(originalRequest);
+      } catch (e) {
+        console.log("Not authorized");
```
Comment (KirillSerg) [inline frontend/src/http/index.ts:36]: better eror message, backend implement it

File: frontend/src/store/services/AuthService.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import api from "../../http";
+import { AxiosResponse } from "axios";
+import {
+  ResponseLogIn,
+  StandartResponse,
+  UserAuthorization,
+} from "../../models/authModels";
+export default class AuthService {
```
Comment (KirillSerg) [inline frontend/src/store/services/AuthService.ts:8]: why class?

File: frontend/src/http/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { AccessToken } from "../models/authModels";
+
+export const API_URL = "http://ec2-52-90-252-13.compute-1.amazonaws.com/api/v1";
+
+const api = axios.create({
+  baseURL: API_URL,
+});
+
+api.interceptors.request.use((config) => {
+  config.headers.Authorization = `Bearer ${localStorage.getItem("access")}`;
```
Comment (KirillSerg) [inline frontend/src/http/index.ts:12]: we can add here test on expired token, to avoid redundant request

**Review by KirillSerg — APPROVED** [review]
big and good work on my opinion

File: frontend/src/store/slices/authSlice.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  isLogged: false,
+  loading: false,
+  error: "",
+};
+
+export const login = createAsyncThunk(
+  "auth/login",
+  async ({ username, password }: UserAuthorization) => {
+    const response = await AuthService.login({ username, password });
+    console.log(response);
```
Comment (salampego) [inline frontend/src/store/slices/authSlice.ts:22]: I think it is better to clean console.log

**Pull request merged**